Assets/Packages/com.studio23.ss2.authsystem.steam/Runtime/Data/SteamAuthProvider.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using Cysharp.Threading.Tasks;
+using Steamworks;
+using Steamworks.Data;
+using Studio23.SS2.AuthSystem.Data;
+using UnityEngine;
+
+namespace Studio23.SS2.AuthSystem.Steam.Data
+{
+    [CreateAssetMenu(fileName = "Steam Auth Provider", menuName = "Studio-23/AuthSystem/Providers/Steam", order = 1)]
+    public class SteamAuthProvider : ProviderBase
+    {
+
+        public uint AppId;
+
+        void OnEnable()
+        {
+            _providerType = ProviderTypes.Steam;
+        }
+
+
+        void OnDisable()
+        {
+            SteamClient.Shutdown();
+        }
+
+        public override async UniTask<int> Authenticate()
+        {
+
+            if (string.IsNullOrEmpty(AppId.ToString()))
+            {
+                Debug.Log("Steam <color=red>App ID not found!</color> Check Resource, Should be at <color=white>Resources/AuthSystem/Providers/SteamAuthProvider.asset</color>");
+                return -1;
+            }
+
+#if !UNITY_EDITOR
+
+            if (SteamClient.RestartAppIfNecessary(AppId))
+            {
+                SteamClient.Shutdown();
+                Application.Quit();
+                return -1;
+            }
+#endif
+            try
+            {
+                SteamClient.Init(AppId, true);
+            }
+            catch (Exception)
+            {
+                SteamClient.Shutdown();
+                Application.Quit();
+                return -1;
+            }
+
+            return 0;
```

Comment:
Instead of hardcoded ints throw exceptions,

Reply:
bad idea to throw exceptions. int returns have meaning in most platforms.